module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -259,10 +263,10 @@ namespace module::vision {
 
                     // DISCARD IF THE DISTANCE FROM THE BALL TO THE ROBOT IS TOO CLOSE
                     // Prevents the robot itself being classed as a ball, ie its arms/hands
-                    if (distance < config.minimum_ball_distance) {
+                    if (angular_distance < config.minimum_ball_distance) {
```

Comment:
Might be better to move the projection/angular distance throwout before this one, otherwise you would need to check that both angular and projection distances are both further away than `config.minimum_ball_distance`.